intrastat_base/intrastat_common.py

```diff
@@ -0,0 +1,167 @@
+                % company_obj.name)
+        if company_obj.currency_id.name != 'EUR':
+            raise Warning(
+                _("The company currency must be 'EUR', but is currently '%s'.")
+                % company_obj.currency_id.name)
+        return True
+
+    @api.one
+    def _check_generate_xml(self):
+        if not self.company_id.partner_id.vat:
+            raise Warning(
+                _("The VAT number is not set for the partner '%s'.")
+                % self.company_id.partner_id.name)
+        return True
+
+    @api.model
+    def _check_xml_schema(self, xml_root, xml_string, xsd_file):
+        '''Validate the XML file against the XSD'''
+        from lxml import etree
+        xsd_etree_obj = etree.parse(
+            tools.file_open(xsd_file))
+        official_schema = etree.XMLSchema(xsd_etree_obj)
+        try:
+            official_schema.assertValid(xml_root)
+        except Exception, e:
+            # if the validation of the XSD fails, we arrive here
+            logger = logging.getLogger(__name__)
+            logger.warning(
+                "The XML file is invalid against the XML Schema Definition")
+            logger.warning(xml_string)
+            logger.warning(e)
+            raise Warning(
+                _("The generated XML file is not valid against the official "
+                    "XML Schema Definition. The generated XML file and the "
+                    "full error have been written in the server logs. "
+                    "Here is the error, which may give you an idea on the "
+                    "cause of the problem : %s.")
+                % str(e))
+        return True
+
+    @api.multi
+    def _attach_xml_file(self, xml_string, declaration_name):
+        '''Attach the XML file to the report_intrastat_product/service
+        object'''
+        self.ensure_one()
+        import base64
+        filename = '%s_%s.xml' % (self.year_month, declaration_name)
+        attach = self.env['ir.attachment'].create({
+            'name': filename,
+            'res_id': self.id,
+            'res_model': self._name,
+            'datas': base64.encodestring(xml_string),
+            'datas_fname': filename})
+        return attach.id
+
+    @api.model
+    def _open_attach_view(self, attach_id, title='XML file'):
+        '''Returns an action which opens the form view of the
+        corresponding attachement'''
+        action = {
+            'name': title,
+            'view_type': 'form',
+            'view_mode': 'form',
+            'res_model': 'ir.attachment',
+            'type': 'ir.actions.act_window',
+            'nodestroy': True,
+            'target': 'current',
+            'res_id': attach_id,
+            }
+        return action
+
+    @api.one
+    def send_reminder_email(self, mail_template_xmlid):
+        mail_template = self.env.ref(mail_template_xmlid)
+        if self.company_id.intrastat_remind_user_ids:
+            self.pool['email.template'].send_mail(
```

Comment:
I'm curious isn't it possible with API 8.0 to do as in event module:
https://github.com/odoo/odoo/blob/8.0/addons/event/event.py#L378

    mail_template.send_mail(self.id)

If not, a comment would be welcome to explain why as this will certainly need to be rewritten in next Odoo version